Sources/AsyncHTTPClient/ConnectionPool/State Machine/HTTPConnectionPool+HTTP1Connections.swift

```diff
@@ -0,0 +1,602 @@
+//
+// This source file is part of the AsyncHTTPClient open source project
+//
+// Copyright (c) 2021 Apple Inc. and the AsyncHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AsyncHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+import XCTest
+
+extension HTTPConnectionPool {
+    /// Represents the state of a single HTTP/1.1 connection
+    private struct HTTP1ConnectionState {
+        enum State {
+            /// the connection is creating a connection. Valid transitions are to: .backingOff, .available and .failed
+            case starting
+            /// the connection is waiting to retry the establishing a connection. Transition to .closed. Must be replaced for a retry.
+            case backingOff
+            /// the connection is available for a new request. Valid transitions to: .leased and .closed
+            case available(Connection, since: NIODeadline)
+            /// the connection is leased and running for a request. Valid transitions to: .available and .closed
+            case leased(Connection)
+            /// the connection is closed. final state.
+            case closed
+        }
+
+        private var state: State
+        private(set) var connectionID: Connection.ID
+        let eventLoop: EventLoop
+
+        init(connectionID: Connection.ID, eventLoop: EventLoop) {
+            self.connectionID = connectionID
+            self.eventLoop = eventLoop
+            self.state = .starting
+        }
+
+        var isConnecting: Bool {
+            switch self.state {
+            case .starting:
+                return true
+            case .backingOff, .closed, .available, .leased:
+                return false
+            }
+        }
+
+        var isBackingOff: Bool {
+            switch self.state {
+            case .backingOff:
+                return true
+            case .starting, .closed, .available, .leased:
+                return false
+            }
+        }
+
+        var isAvailable: Bool {
+            switch self.state {
+            case .available:
+                return true
+            case .backingOff, .starting, .leased, .closed:
+                return false
+            }
+        }
+
+        var isLeased: Bool {
+            switch self.state {
+            case .leased:
+                return true
+            case .backingOff, .starting, .available, .closed:
+                return false
+            }
+        }
+
+        var availableSince: NIODeadline? {
+            switch self.state {
+            case .available(_, since: let lastReturn):
+                return lastReturn
+            case .backingOff, .starting, .leased, .closed:
+                return nil
+            }
+        }
+
+        mutating func started(_ connection: Connection) {
```

Comment:
"started" makes me think that an attempt to establish a connection has started rather than the connection has been established and is ready to use. "connected" may be slightly clearer here.